components/script/dom/websocket.rs

```diff
@@ -201,20 +201,30 @@ impl WebSocketMethods<crate::DomTypeHolder> for WebSocket {
         protocols: Option<StringOrStringSequence>,
     ) -> Fallible<DomRoot<WebSocket>> {
         // Steps 1-2.
-        let url_record = ServoUrl::parse(&url).or(Err(Error::Syntax))?;
+        let mut url_record = ServoUrl::parse(&url).or(Err(Error::Syntax))?;
 
-        // Step 3.
+        // Step 3-4-5.
         match url_record.scheme() {
             "ws" | "wss" => {},
+            "http" => {
+                url_record
+                    .as_mut_url()
+                    .set_scheme("ws")
+                    .expect("Can't set scheme from http to ws");
+            },
+            "https" => {
+                url_record
+                    .as_mut_url()
+                    .set_scheme("wss")
+                    .expect("Can't set scheme from https to wss");
+            },
             _ => return Err(Error::Syntax),
         }
 
-        // Step 4.
         if url_record.fragment().is_some() {
             return Err(Error::Syntax);
         }
 
-        // Step 5.
```

Comment:
Instead of removing these spec comments, can you update them appropriately? For example,

```
        if url_record.fragment().is_some() {
            return Err(Error::Syntax);
        }
```
seems to be step 7 now. (and so on for the rest of the function)

BTW, the usual format for spec comments is to include the spec-text, like this:
```
Step 7. If urlRecord’s fragment is non-null, then throw a "SyntaxError" DOMException.
```
That is preferrable since spec numbers can change over time. But this function didn't do that before either, so just updating the numbers is fine.

Reply:
Nice suggestion! Thanks. I fixed. However, I'm not sure about step 10 and 11 in the specs.